src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    static final int MAX_RANDOM_VALUE = 50;
+    static final int DEFAULT_FIGURE_VALUE = 10;
+    static final String DEFAULT_FIGURE_COLOR = Color.WHITE.name();
+    private final Random rnd = new Random();
+    private final ColorSupplier color = new ColorSupplier();
+
+    private int getRandomFigureValue() {
+        return rnd.nextInt(MAX_RANDOM_VALUE) + 1;
+    }
+
+    public Figure getRandomFigure() {
+        switch (FigureType.values()[rnd.nextInt(FigureType.values().length)]) {
+            case SQUARE:
+                int squareSide = getRandomFigureValue();
+                String figureColor = color.getRandomColor();
+                return new Square(squareSide, figureColor);
```

Comment:
```suggestion
            case SQUARE:
                return getRandomSquare();
```
lets add private methods to get each random figure - this would make your code easier to read

Reply:
yeah of course ) thank you. do I have to repush it after your approval?